spec/function.dd

```diff
@@ -2524,6 +2524,288 @@ $(H3 $(LNAME2 this-reference, `this` Reference))
         )
 
 
+
+$(H2 $(LNAME2 refscopereturn, Ref Scope Return Cases))
```

Comment:
It would be nice to give this a proper place and introduction instead of just dropping it in the middle of function.dd, which is already pretty big.

Reply:
The whole spec could us a refactoring. The structure of it is very old. But this is properly a separate issue.